sdks/python/src/opik/integrations/harbor/opik_tracker.py

```diff
@@ -0,0 +1,473 @@
+        if result.verifier_result and result.verifier_result.rewards:
+            # Get error message if available
+            error_msg = getattr(result.verifier_result, "error", None) or getattr(
+                result, "error", None
+            )
+            feedback_scores = _rewards_to_feedback_scores(
+                result.verifier_result.rewards, error=error_msg
+            )
+
+        opik_context.update_current_trace(
+            output=output_dict,
+            feedback_scores=feedback_scores,
+        )
+
+        # Link to experiment
+        trace_data = context_storage.get_trace_data()
+        if trace_data is not None:
+            service = experiment_service.get_service()
+            LOGGER.debug(
+                "Linking trial to experiment: trial=%s, trace_id=%s, service=%s",
+                config.trial_name,
+                trace_data.id,
+                service,
+            )
+            if service is not None:
+                source = getattr(config.task, "source", None)
+                task_name = (
+                    config.task.name
+                    if hasattr(config.task, "name")
+                    else str(config.task.path)
+                )
+                service.link_trial(
+                    trial_name=config.trial_name,
+                    trace_id=trace_data.id,
+                    source=source,
+                    task_name=task_name,
+                    agent_name=config.agent.name,
+                    model_name=getattr(config.agent, "model_name", None),
+                )
+            else:
+                LOGGER.debug(
+                    "No experiment service available, skipping experiment linking"
+                )
+
+        return result
+
+    wrapped.opik_tracked = True  # type: ignore[union-attr]
+    return wrapped
+
+
+def _wrap_setup_environment(
+    original: Callable, project_name: Optional[str]
+) -> Callable:
+    """Wrap Trial._setup_environment with tracing."""
+
+    @track(name="setup_environment", tags=["harbor"], project_name=project_name)
+    @functools.wraps(original)
+    async def wrapped(self: "Trial") -> None:
+        opik_context.update_current_span(
+            input={"phase": "environment_setup"},
+            metadata={"created_from": "harbor"},
+        )
+        await original(self)
+        opik_context.update_current_span(output={"status": "completed"})
+
+    wrapped.opik_tracked = True  # type: ignore[union-attr]
+    return wrapped
+
+
+def _wrap_setup_agent(original: Callable, project_name: Optional[str]) -> Callable:
+    """Wrap Trial._setup_agent with tracing."""
+
+    @track(name="setup_agent", tags=["harbor"], project_name=project_name)
+    @functools.wraps(original)
+    async def wrapped(self: "Trial") -> None:
+        opik_context.update_current_span(
+            input={"phase": "agent_setup"},
+            metadata={"created_from": "harbor"},
+        )
+        await original(self)
```

Comment:
even if it returns None today, we should still do
```python
result = original(self)  # or result = await original(self)
return result
```
otherwise our code will break theirs if one day they decide to add a return value (especially relevant for private functions)

It's relevant for other wrappers too.